src/ORM/ArrayList.php

```diff
@@ -572,6 +572,18 @@ public function sort()
         return $list;
     }
 
+    /**
+     * Shuffle the items in this array list
+     *
+     * @return $this
+     */
+    public function shuffle()
+    {
+        shuffle($this->items);
+
+        return $this;
```

Comment:
Some of the other methods in this class are immutable, do you want this to be as well?

Reply:
Given the nature of PHP's `shuffle` - I don't really think it should be. Happy to adjust if asked.

Reply:
That's the whole point of this function - to reorder in place, without allocating extra memory. Otherwise we could do 
```php
$array = $arrayList->toArray();
shuffle($array);
return new ArrayList($array);
```